### PR TITLE
Dialog Evolution Doc Updates

### PR DESCRIPTION
**Pull request by Aevyz**
Right, hopefully I did not lose half of botboys changes like the last attempt...

Documentation builds at RD https://roboydialog.readthedocs.io/en/memory-doc/user_manual/1_getting_started.html no problem. 

#59 should be completely incorporated now, minus direct DB link.

https://github.com/Roboy/roboy_memory/pull/43

Todos:

- [x] Spell Check
- [x] Resolve Merge Conflict
- ~~[ ] Add page on expected outcomes~~
- [x] Add links (done as soon as I am on laptop)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,5 @@ resources/word2vec_toy_data_and_model
 resources_nlu/word2vec/roboy_enwik8.txt
 
 docs/doxyxml
+
+dialog/
```
Comment (missxa) [inline .gitignore]: why is it ignored??

File: README.md [hunk 2]
```diff
@@ -1,33 +1,200 @@
...
+
+## What is this Project
+
+This repository contains a dialog system developed for the humanoid robot [Roboy](roboy.org).
+
+## Does This Project Have Documentation
+
+Yes, you can find them [here](https://roboy-dialog.readthedocs.io/).
+
+## How does the Dialog System Work
```
Comment (missxa) [inline README.md]: lets ditch this section from readme

File: README.md [hunk 3]
```diff
@@ -1,33 +1,200 @@
...
-There are interfaces for each step in the processing pipeline to enable an easy exchange of elements. The goal would be to easily exchange personalities based on the occasion.
+### Quick-Install
 
-The implementation of the pipeline is in Java. Integrations with tools in other languages, like C++ RealSense stuff, should be wrapped in a module in the pipeline.
+The quick guide for people who don't care about any of the technical mumbo jumbo. This is a local installation guide.
+
+[Set Environmental Variables](#environmental-variables). **Only do this once.**
 
-## How to run it
+```bash
```
Comment (missxa) [inline README.md]: so this is a very nice chunk of bash script, but you gotta explain what each line does: either separate it to steps or add a `#` in the end of each line with a short explanation

File: README.md [hunk 4]
```diff
@@ -1,33 +1,200 @@
...
+- Working Internet Connection for downloading dependencies
+
+### Recommendations
+
+- Intellij
+- At least 8GB of RAM
+    - This project will run with a 4GB of RAM, however development will be a horror-show.
+- Ubuntu (or variation) 16.04 LTS
+    - Needed for `ROS Kinetic`
+    - If you do not need ROS features, any version of Linux should do
```
Comment (missxa) [inline README.md]: macOS works too

File: README.md [hunk 5]
```diff
@@ -1,33 +1,200 @@
...
+- At least 6GB of Disk Space
+    - ~ 4GB for Maven Dependencies
+    - ~ 500MB for Roboy_Dialog with all sub-modules and files generated via mvn clean install
+    - Rest is a ballpark estimate for Neo4J, Redis, ROS and their dependencies
+
+> Note that `apt` is bugged with Ubuntu 16.04. See: https://bugs.launchpad.net/ubuntu/+source/appstream/+bug/1583845 if you get problems with `apt-update`.
+
+### Optional
```
Comment (missxa) [inline README.md]: it's confusing that optional and Installing and Usage via Command Line are the same kind of heading. 
would make more sense to push optional and recommendations to be subsections of requirements

File: README.md [hunk 6]
```diff
@@ -1,33 +1,200 @@
...
+
+- ROS
+- Redis
```
Comment (missxa) [inline README.md]: not used by anyone. we shouldnt mention it

File: README.md [hunk 7]
```diff
@@ -36,57 +203,52 @@ export ROS_MASTER_URI=http://rosmaster:11311
...
+export ROS_HOSTNAME="127.0.0.1"
+export NEO4J_ADDRESS="bolt://127.0.0.1:7687"
+export NEO4J_USERNAME="neo4j"
+export NEO4J_PASSWORD="neo4jpassword"
+export REDIS_URI="redis://localhost:6379/0"
+export REDIS_PASSWORD="root"
 ```
 
-#### Using the Google Word2Vec Model in NLU
+## Configuration of Roboy_Dialog
```
Comment (missxa) [inline README.md]: of roboy_dialog or Roboy Dialog

File: dialog/src/main/java/roboy/dialog/ConversationManager.java [hunk 8]
```diff
@@ -82,63 +82,40 @@ public static void main(String[] args) throws IOException {
         analyzers.add(new ProfanityAnalyzer());
 
 
-        //Roboy mode: Repeat a conversation a few times.
-        if(ConfigManager.ROS_ENABLED) {
+        //I/O specific core behaviour
```
Comment (missxa) [inline dialog/src/main/java/roboy/dialog/ConversationManager.java]: are these your changes @Aevyz 
seems like botboys

File: docs/Usage/9_Troubleshooting.rst [hunk 9]
```diff
@@ -0,0 +1,29 @@
+Troubleshooting
```
Comment (missxa) [inline docs/Usage/9_Troubleshooting.rst]: please add a link to issues page of the github repo as:
a) a place to look for solutions as well
b) post new issues

File: docs/Usage/0_installation.rst [hunk 10]
```diff
@@ -1,52 +1,26 @@
...
 
-Now, import Dialog System as a Maven project into the IDE of your choice. Build and execute using ``roboy.dialog.ConversationManager`` as the main class.
 
-Using a telegram bot for I/O
-============================
+.. _Optional steps:
 
-Register a bot as described on the `telegram website`_.
-.._telegram website: https://core.telegram.org/bots#3-how-do-i-create-a-bot
+Optional steps
```
Comment (missxa) [inline docs/Usage/0_installation.rst]: imo the installation of the telegram API should be moved to the tutorials and it is not an optionall installation step @SezerCan @polinominom @Mireu

**Review by missxa — APPROVED** [review]
minor fixes, otherwise good

File: docs/Usage/0_installation.rst (hunk 10 above)
Comment (Mireu) [reply docs/Usage/0_installation.rst]: Moved it to User Manual -> Configuration in b4971d9c369e22b2d2f0d77b649c34b6542891f6 .

File: docs/Usage/9_Troubleshooting.rst (hunk 9 above)
Comment (Aevyz) [reply docs/Usage/9_Troubleshooting.rst]: Is the current addendum satisfactory? Also Roboy_Parser has its issues page disabled, is that on purpose?

**Review by josephbirkner — APPROVED** [review]

**Pull request merged**